DPL CCDB: handle the Cache-Valid-Until header

ktf opened the pull request:
DPL CCDB: handle the Cache-Valid-Until header

This will not recheck the etag and consider the cache valid until
the timestamp in milliseconds in the Cache-Valid-Until is not passed
by the data timestamp.

ktf commented:
@costing @shahor02 attempt at supporting Cache-Valid-Until header in the CCDB Reply. One thing it's not clear to me is the unit of Cache-Valid-Until. Is it milliseconds?

shahor02 commented on Framework/CCDBSupport/src/CCDBHelpers.cxx:
```diff
@@ -217,7 +218,9 @@ auto populateCacheWith(std::shared_ptr<CCDBFetcherHelper> const& helper,
-      checkValidity = std::abs(int(timingInfo.tfCounter - url2uuid->second.lastCheckedTF)) >= chRate;
+      // We check validity every chRate timeslices or if the cache is expired
+      uint64_t validUntil = url2uuid->second.cacheValidUntil;
+      checkValidity = (std::abs(int(timingInfo.tfCounter - url2uuid->second.lastCheckedTF)) >= chRate) || (validUntil <= timestamp);
```
Should not it be `&& (validUntil <= timestamp)` ?

shahor02 replied:
@ktf my yesterday fix was not concerning this `||` but line before it: https://github.com/AliceO2Group/AliceO2/pull/12885/files

shahor02 commented on Framework/CCDBSupport/src/CCDBHelpers.cxx:
```diff
@@ -240,6 +250,7 @@ auto populateCacheWith(std::shared_ptr<CCDBFetcherHelper> const& helper,
       helper->mapURL2UUID[path].lastCheckedTF = timingInfo.tfCounter;
       if (etag.empty()) {
         helper->mapURL2UUID[path].etag = headers["ETag"]; // update uuid
+        helper->mapURL2UUID[path].cacheValidUntil = headers["Cache-Valid-Until"].empty() ? 0 : std::stoul(headers["Cache-Valid-Until"]);
```
It appears  that this `"Cache-Valid-Until"` is provided not at the 1st query (with empty etag) and at the 1st query with valid etag (actually at the 2nd one, see my email to @costing ). 
So, there is no point in its caching here, instead should be cached as done here https://github.com/AliceO2Group/AliceO2/blob/956b26710723080d559b30835eb15cadaba0d503/Framework/CCDBSupport/src/CCDBHelpers.cxx#L265 and also here https://github.com/AliceO2Group/AliceO2/blob/956b26710723080d559b30835eb15cadaba0d503/Framework/CCDBSupport/src/CCDBHelpers.cxx#L275 as 
```
else {
  helper->mapURL2UUID[path].cacheValidUntil = headers["Cache-Valid-Until"].empty() ? 0 : std::stoul(headers["Cache-Valid-Until"]);
}
```

ktf replied:
@shahor02 updated.

ktf committed:
DPL CCDB: handle the Cache-Valid-Until header

This will not recheck the etag and consider the cache valid until
the timestamp in milliseconds in the Cache-Valid-Until is not passed
by the data timestamp.

ktf commented:
Passes the full system test.
